### PR TITLE
Fixes to keep caps

### PR DESCRIPTION
**Pull request by SermishaNarayana**
Closes # (if applicable).

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [ ] Code and workflow changes are sufficiently documented.
- [ ] Changed dependencies are added to `envs/environment.yaml`.
- [ ] Changes in configuration options are added in all of `config.default.yaml`.
- [ ] Changes in configuration options are also documented in `doc/configtables/*.csv`.
- [ ] A release note `doc/release_notes.rst` is added.

### CODE REVIEW

File: scripts/prepare_sector_network.py [hunk 1]
```diff
@@ -851,19 +851,18 @@ def annuity_factor(v):
...
-def add_generation(n, costs):
+def add_generation(n, costs, existing_capacities, existing_generation):
     logger.info("Adding electricity generation")
 
     nodes = pop_layout.index
 
     fallback = {"OCGT": "gas"}
     conventionals = options.get("conventional_generation", fallback)
-
+
```
Comment (martacki) [inline scripts/prepare_sector_network.py:861]:
```suggestion
```

Comment (martacki) [inline scripts/prepare_sector_network.py:861]:
```suggestion

```

File: scripts/prepare_sector_network.py [hunk 2]
```diff
@@ -4002,6 +4020,7 @@ def get_capacities_from_elec(n, carriers, component):
     )
     pop_weighted_energy_totals.update(pop_weighted_heat_totals)
 
+
```
Comment (martacki) [inline scripts/prepare_sector_network.py:4023]:
```suggestion
```

File: scripts/add_electricity.py [hunk 3]
```diff
@@ -419,12 +419,20 @@ def attach_wind_and_solar(
             else:
                 capital_cost = costs.at[car, "capital_cost"]
 
+            if not ppl.query("carrier == @car").empty:
+                caps = ppl.query("carrier == @car").groupby("bus").p_nom.sum()
+                caps = pd.Series(data = caps, index = ds.indexes["bus"]).fillna(0)
+            else:
+                caps = pd.Series(index = ds.indexes["bus"]).fillna(0)
+
```
Comment (martacki) [inline scripts/add_electricity.py:427]: What do these changes have to do with keeping existing caps in the sector model?

Comment (SermishaNarayana) [reply scripts/add_electricity.py:427]: Ah no! These are for adding renewables from custom_powerplants.csv. Sorry, I thought I had mentioned that in the commit message

File: scripts/prepare_sector_network.py [hunk 4]
```diff
@@ -874,9 +872,28 @@ def add_generation(n, costs):
...
+            p_nom_extendable=(
+                True
+                if generator
+                in snakemake.params.electricity.get("extendable_carriers", dict()).get(
+                    "Generator", list()
+                )
+                else False
+            ),    
+            p_nom=(
+                existing_capacities[generator] / costs.at[generator, "efficiency"]
```
Comment (yerbol-akhmetov) [inline scripts/prepare_sector_network.py:884]: Here it needs to be `existing_capacities[generator] / existing_efficiencies[generator]`. As I have suggested previosuly in the chat.

File: scripts/prepare_sector_network.py [hunk 5]
```diff
@@ -851,19 +851,17 @@ def annuity_factor(v):
     return costs
 
 
-def add_generation(n, costs):
+def add_generation(n, costs, existing_capacities, existing_generation):
```
Comment (yerbol-akhmetov) [inline scripts/prepare_sector_network.py:854]: Here shouldn't be existing_efficiencies as a second input?

File: scripts/add_electricity.py (hunk 3 above)
Comment (martacki) [reply scripts/add_electricity.py:427]: Ah, I see. Ok! I guess we can keep it in the same PR, it's not too different. Thanks!

**Review by martacki — APPROVED** [review]
LGTM

File: scripts/prepare_sector_network.py (hunk 4 above)
Comment (martacki) [reply scripts/prepare_sector_network.py:884]: If you want you can directly make a code suggestion, makes it easier to apply your changes

File: scripts/prepare_sector_network.py (hunk 5 above)
Comment (SermishaNarayana) [reply scripts/prepare_sector_network.py:854]: Ah yes! My bad... I'll change that

**Comment by yerbol-akhmetov** [comment]
LGTM!

**Review by yerbol-akhmetov — APPROVED** [review]

**Pull request merged**